### PR TITLE
SISRP-23223 - CS Link API - suppress `properties` hash in front-end JSON

### PR DESCRIPTION
**Pull request by dfkaye**
https://jira.berkeley.edu/browse/SISRP-23223

### CODE REVIEW

File: app/models/campus_solutions/link.rb [hunk 1]
```diff
@@ -46,11 +46,13 @@ def get_url(url_id, placeholders = {})
             end
           end
 
+          # promote properties we're interested in, delete the rest
           link[:properties].each do |property|
             if property[:name] == 'NEW_WINDOW' && property[:value] == 'Y'
               link[:showNewWindow] = true
             end
           end
+          link.delete(:properties)
```
Comment (johncrossman) [inline app/models/campus_solutions/link.rb]: @dfkaye @raydavis  I believe this introduces a bug. Deleting `:properties` this hash deletes it from the cached data. I.e., the next time you hit this feed you will find no `:properties` and you'll get error: `undefined method 'each' for nil:NilClass`

Comment (johncrossman) [inline app/models/campus_solutions/link.rb]: @dfkaye I'll create a Jira. The good news: As explained to me by @raydavis, no such bug will happen in dev, qa and prod where memcached is in effect.  The hash delete will not update / modify what is in memcached. Thus, a new QA PR is not needed. Agreed, @raydavis?

Comment (raydavis) [inline app/models/campus_solutions/link.rb]: I believe so, @johncrossman , but I have to admit I haven't verified with a test. I'll do that locally now.

And THANK YOU for noticing that subtle-but-deadly bug.

Comment (johncrossman) [inline app/models/campus_solutions/link.rb]: @raydavis @dfkaye  I created https://jira.berkeley.edu/browse/SISRP-23228 and put it in the next sprint (GL 6.0d release). Feel free to reprioritize.

Comment (raydavis) [inline app/models/campus_solutions/link.rb]: Confirmed.

```
WITH DEFAULT RAILS CACHE:

>> original_hash = {this: 'thing', that: 'other'}
{:this=>"thing", :that=>"other"}
>> Rails.cache.write('THE KEY', original_hash)
true
>> fetched_hash = Rails.cache.read 'THE KEY'
{:this=>"thing", :that=>"other"}
>> fetched_hash.delete(:that)
"other"
>> another_fetched_hash = Rails.cache.read 'THE KEY'
{:this=>"thing"}

WITH MEMCACHED:

>> original_hash = {this: 'thing', that: 'other'}
{:this=>"thing", :that=>"other"}
>> Rails.cache.write('THE KEY', original_hash)
72057594037927936
>> fetched_hash = Rails.cache.read 'THE KEY'
{:this=>"thing", :that=>"other"}
>> fetched_hash.delete(:that)
"other"
>> another_fetched_hash = Rails.cache.read 'THE KEY'
{:this=>"thing", :that=>"other"}
```

**Pull request merged**